clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -3872,6 +3872,13 @@ LangAS CodeGenModule::GetGlobalVarAddressSpace(const VarDecl *D) {
     if (!D || D->getType().getAddressSpace() == LangAS::Default) {
       return LangAS::opencl_global;
     }
+    if (D)
+      AddrSpace = D->getType().getAddressSpace();
+    if (AddrSpace == LangAS::opencl_private ||
+        AddrSpace == LangAS::opencl_local)
+      // SYCL explicit SIMD path: recognize globals in private or local address
+      // space.
+      return AddrSpace;
```

Comment:
You don't have a check in codegen test that these address spaces are actually preserved and not replaced with something else by `return getTargetCodeGenInfo().getGlobalVarAddressSpace(*this, D);`

Reply:
why would I need that test?

Reply:
Apparently, you need this particular change here which changes an address space of some variables. However, I don't see any tests that check for particular address space

Reply:
@DenisBakhvalov  agreed to help here with the test

Reply:
Apparently, we found that this check is not needed anymore. Removed.